src/c/coap-server.c

```diff
@@ -0,0 +1,465 @@
+/* CoAP server for device-coap-c
+ *
+ * Copyright (C) 2018 Olaf Bergmann <bergmann@tzi.org>
+ * Copyright (c) 2020 Ken Bannister
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <float.h>
+#include <signal.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netdb.h>
+
+#include <coap2/coap.h>
+#include "edgex/devices.h"
+#include "device-coap.h"
+
+/* Maximum length of a string containing numeric values. */
+#define INT32_STR_MAXLEN 10
+#define FLOAT64_STR_MAXLEN (DBL_MAX_10_EXP + 2)
+
+#define RESOURCE_SEG1 "a1r"
+#define MSG_PAYLOAD_INVALID "payload not valid"
+#define MEDIATYPE_TEXT_PLAIN "text/plain"
+#define MEDIATYPE_APP_JSON "application/json"
+#define CONTENT_FORMAT_UNDEFINED UINT16_MAX
+
+static coap_driver *sdk_ctx;
+
+/* controls input loop */
+static int quit = 0;
+
+/* signal handler for input loop */
+static void
+handle_sig (int signum)
+{
+  (void)signum;
+  quit = 1;
```

Comment:
It is better to handle various signals like SIGTERM, SIGSEGV etc

Reply:
device-coap-c handles SIGINT and SIGTERM. You can see the setup for the handlers near the end of `main()`. I think handling those two signals is sufficient. A review of device-bacnet-c, device-grove-c, and the examples in device-sdk-c shows they all handle SIGINT. device-grove-c also handles SIGTERM. I don't see handling for any other signals.

It might be easier to discuss on a per-signal basis for any specific concerns.